process.go

```diff
@@ -337,18 +337,35 @@ func (p *process) eventLoop(ctx context.Context, group *errgroup.Group) error {
 
 			ctrl := p.getPeerController(ctx, chanInfo.peer, group.Go)
 
-			// Lookup the outgoing peer to supplement the
-			// information on the resolved event.
+			// Lookup the outgoing peer to supplement the information on the
+			// resolved event. Here we handle a channel lookup error
+			// differently to the incoming channel, because it's possible
+			// we were forwarded a HTLC with a bogus outgoing channel. If
+			// this is the case, LND would have failed the HTLC back even if
+			// we let it through. We catch and log that error, rather than
+			// exiting like we do with incoming channels (where we reasonably
+			// expect to find the channel). We still enforce channel lookup
+			// for successful HTLCs, because then we know that the channel
+			// does exist and should be found.
+			var outgoingPeer route.Vertex
```

Comment:
Instead of an all-zeroes key in the database, a NULL value would be cleaner I think.

Reply:
Happy for me to add a migration to relax the not-null restriction on `outgoing_peer` to do this?

Reply:
An alternative would be to just not record anything in the fwd log at all in these cases?